### PR TITLE
Add Cursor tests and fix some bugs

### PR DESCRIPTION
**Pull request by hughbe**
###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/1527)

### CODE REVIEW

File: src/System.Windows.Forms.Design/tests/UnitTests/System/ComponentModel/Design/DesignerHostTests.cs [hunk 1]
```diff
@@ -883,7 +885,7 @@ public void DesignerHost_Add_DuringUnload_ThrowsException()
             Assert.Empty(surface.LoadErrors);
             Assert.False(host.Loading);
         }
-
+ */
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/tests/UnitTests/System/ComponentModel/Design/DesignerHostTests.cs]: Please disable it via the `Skip` property, e.g. `[Fact(Skip = "Unstable test, https://github.com/dotnet/winforms/issues/1460")]`

Comment (RussKie) [reply src/System.Windows.Forms.Design/tests/UnitTests/System/ComponentModel/Design/DesignerHostTests.cs]: Actually don't worry about this, I'm going to fix it in separate commit

**Comment by RussKie** [comment]
Please remove the last commit

File: src/System.Windows.Forms/src/System/Windows/Forms/Cursors.cs [hunk 2]
```diff
@@ -4,452 +4,104 @@
...
-
-        private Cursors()
-        {
-        }
+        private static Cursor s_appStarting;
+        private static Cursor s_arrow;
+        private static Cursor s_cross;
+        private static Cursor s_defaultCursor;
+        private static Cursor s_iBeam;
+        private static Cursor no;
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Cursors.cs]: Any reason why this and few others below weren't changed?

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/CursorsTests.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            yield return new object[] { I(() => Cursors.SizeNWSE) };
+            yield return new object[] { I(() => Cursors.SizeWE) };
+            yield return new object[] { I(() => Cursors.UpArrow) };
+            yield return new object[] { I(() => Cursors.VSplit) };
+            yield return new object[] { I(() => Cursors.WaitCursor) };
+        }
+
+        [Theory]
+        [MemberData(nameof(Cursors_TestData))]
+        public void Cursors_KnownCursor_Get_ReturnsExpected(Func<Cursor> cursorFactory)
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/CursorsTests.cs]: The follow is probably more fitting:
```suggestion
        public void Cursors_KnownCursor_Get_ReturnsExpected(Func<Cursor> getCursor)
```

File: src/System.Windows.Forms/src/System/Windows/Forms/CursorConverter.cs [hunk 4]
```diff
@@ -87,12 +87,7 @@ public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo c
         /// </summary>
         public override object ConvertTo(ITypeDescriptorContext context, CultureInfo culture, object value, Type destinationType)
         {
-            if (destinationType == null)
-            {
-                throw new ArgumentNullException(nameof(destinationType));
-            }
-
-            if (destinationType == typeof(string) && value != null)
+            if (destinationType == typeof(string) && value is Cursor cursorValue)
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/CursorConverter.cs]: Can we do the following to aling with the similar code in `CovertTo`?
```suggestion
            if (destinationType == typeof(string) && value is Cursor cursor)
```

File: src/System.Windows.Forms/src/System/Windows/Forms/CursorConverter.cs [hunk 5]
```diff
@@ -142,15 +137,11 @@ public override object ConvertTo(ITypeDescriptorContext context, CultureInfo cul
...
                 {
-                    MemoryStream ms = new MemoryStream();
-                    Cursor cursor = (Cursor)value;
-                    cursor.SavePicture(ms);
-                    ms.Close();
-                    return ms.ToArray();
+                    return cursor.GetData();
                 }
-                else
+                else if (value == null)
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/CursorConverter.cs]: `else` is redundant
Arguably null check should occur before the cast.

Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/CursorConverter.cs]: Unfortunately not: consider the case of `ConvertTo(new object(), typeof(byte[]))`.

I'd expect an exception to be thrown (NotSupportedException from the base class). Previously, we incorrectly threw `InvalidCastException`

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DataGridViewCellStyleTests.cs [hunk 6]
```diff
@@ -1139,18 +1141,15 @@ public static IEnumerable<object[]> ToString_TestData()
                     Tag = "tag",
                     WrapMode = DataGridViewTriState.True
                 },
-                "DataGridViewCellStyle { BackColor=Color [Red], ForeColor=Color [Blue], SelectionBackColor=Color [Green], SelectionForeColor=Color [Yellow], Font=[Font: Name=Microsoft Sans Serif, Size=8.25, Units=3, GdiCharSet=1, GdiVerticalFont=False], NullValue=null, DataSourceNullValue=dbNull, Format=format, WrapMode=True, Alignment=BottomCenter, Padding={Left=1,Top=2,Right=3,Bottom=4}, Tag=tag }"
+                $"DataGridViewCellStyle {{ BackColor=Color [Red], ForeColor=Color [Blue], SelectionBackColor=Color [Green], SelectionForeColor=Color [Yellow], Font={font}, NullValue=null, DataSourceNullValue=dbNull, Format=format, WrapMode=True, Alignment=BottomCenter, Padding={{Left=1,Top=2,Right=3,Bottom=4}}, Tag=tag }}"
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DataGridViewCellStyleTests.cs]: Blobs are easier to verify with [ApprovalTests](https://github.com/approvals/ApprovalTests.Net)

**Comment by RussKie** [comment]
Please amalgamate the last commit and it is ready to go

**Pull request merged**